### PR TITLE
[ENH] Implement find_replace API change

### LINKED ISSUE #255: [ENH] find_replace on multiple columns
modify find_replace to be applicable to a list of columns

### PR DESCRIPTION
**Pull request by ericmjl**
This PR now lets us `find_replace` using the following pattern:

```python
df = pd.DataFrame(...)

df = df.find_replace(column1={"hello": "world"}, column2={"what's up": "world"})
```

I have left out `find_replace_all`, because `df.replace()` natively fulfills this functionality.

This PR closes #255.

h/t @nickdelgrosso for the API change suggestion!

### CODE REVIEW

**Comment by ericmjl** [comment]
Requested reviews from those who recently edited the files touched (based on GitHub's suggestions).

File: janitor/functions.py [hunk 1]
```diff
@@ -3003,27 +2997,64 @@ def find_replace(
...
+        df = df.find_replace(
+            match='regex',
+            order={'coffee$': 'latte'},
+        )
+
+    To perform a find and replace on the entire dataframe,
+    pandas' ``df.replace()`` function provides the appropriate functionality.
+    You can find more detail on the replace_ docs.
+
+    .. _replace: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.replace.html  # noqa: E501
```
Comment (hectormz) [inline janitor/functions.py]: Do you want to move `# noqa: E501` to the end of the docstring (line 3040) like #593

File: tests/functions/test_find_replace.py [hunk 2]
```diff
@@ -50,6 +50,4 @@ def test_find_replace_regex(df_orders):
 @pytest.mark.functions
 def test_find_replace_regex_match_rases_error(df_orders):
```
Comment (hectormz) [inline tests/functions/test_find_replace.py]: Can we change `rases` to `raises` in test name?

File: janitor/functions.py (hunk 1 above)
Comment (ericmjl) [reply janitor/functions.py]: Yes, good catch. I have been sloppy about this. This is even I have to have my code reviewed :smile: (I probably sound like a broken tape recorder by now.) Let me fix it.

File: tests/functions/test_find_replace.py (hunk 2 above)
Comment (ericmjl) [reply tests/functions/test_find_replace.py]: Good catch.

**Comment by hectormz** [comment]
@ericmjl If I'm looking at this correctly, this will require column names that are provided to be capable of being variable names/keyword arguments? So a column `"num_dice"` or `"numdice"` can be provided, but not `"num dice"`? If someone has incompatible column names, they should use the underlying `_find_replace()`?

**Comment by ericmjl** [comment]
> @ericmjl If I'm looking at this correctly, this will require column names that are provided to be capable of being variable names/keyword arguments? 

Yes.

> So a column `"num_dice"` or `"numdice"` can be provided, but not `"num dice"`? 

I think that is right. I will have to check what happens, and should probably add a test.

> If someone has incompatible column names, they should use the underlying `_find_replace()`?

Probably yes, and this behaviour should be documented.

------

@hectormz thanks for catching these issues! It's very helpful. Can you request changes on this, so that I don't accidentally merge?

**Comment by hectormz** [comment]
@ericmjl done! I'm guessing Python just won't parse the keyword, so we probably just need to add documentation.

**Comment by ericmjl** [comment]
@hectormz thanks for the good suggestion. Pushed the change up, and I'm going to merge this in.

**Pull request merged**